metalgpy/_sample.py

```diff
@@ -36,12 +38,104 @@ def list_type(list_exp):
             return LIST_KCATEGORICAL
 
 class BaseSampler:
-    def __init__(self, exp, distributions: dict=None, rng:np.random.RandomState=None):
+    """Defines a Base Sampler
+    
+    Args:
+        exp (Expression): a metalgpy expression.
+        distributions (dict, optional): preferred distributions for drawing samples. Defaults to None.
+        rng (int, optional): a random seed to generate same stream of values.Defaults to None.
+    """
+    def __init__(self,
+                 exp,
+                 distributions: dict=None,
+                 rng:np.random.RandomState=None):
         self.exp = exp
-        self.distributions = distributions
+        self.distributions = self._map_dist(distributions)
         self.rng = rng
+        self._cache = {}
+
+    def sample(self, size=None, to_nd_array=True):
```

Comment:
That makes sense. In essence I had the same concern about computational efficiency, as to what would be an efficient way to return the samples if it categorical variables are passed because we can't return an nd-array. I'll make this change.